src/components/Dnsrecon/Dnsrecon.tsx

```diff
@@ -1,123 +1,208 @@
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
 import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+import { RenderComponent } from "../UserGuide/UserGuide";
 
-const title = "DNSRecon";
-const description_userguide =
-    "DNSRecon is a python script that has an extensive list of functions. This tool is primarily used for " +
-    "DNS enumeration and scanning, for example, it may enumerate DNS records, SRV records, and hosts and " +
-    "domains using Google.\n\nFurther uses for the tool can be found at: https://www.kali.org/tools/dnsrecon/\n\n" +
-    "Using DNSRecon:\n" +
-    "Step 1: Enter a target domain URL, for example, https://www.deakin.edu.au\n\n" +
-    "Step 2: Click scan to commence DNSRecon's operation.\n\n" +
-    "Step 3: View the output block below to view the results of the tool's execution.";
-
-interface FormValues {
+/**
+ * Represents the form values for the Dnsrecon component.
+ */
+interface FormValuesType {
     url: string;
 }
 
-export function Dnsrecon() {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
+/**
+ * The Dnsrecon component.
+ * @returns The Dnsrecon component.
+ */
+function Dnsrecon() {
+    // Component State Variables.
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable to check if the installation modal is open.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state for the installation modal.
 
-    let form = useForm({
+    // Component Constants.
+    const title = "DNSRecon"; // Title of the component.
+    const description = "DNSRecon is a tool for DNS enumeration and scanning."; // Description of the component.
+    const steps =
+        "Step 1: Enter a target domain URL, for example, https://www.deakin.edu.au\n" +
+        "Step 2: Click scan to commence DNSRecon's operation.\n" +
```

Comment:
Change scan to "Start DNSRecon" to make it consistent with the button text. 

- [ ] Change text to Start DNSRecon.